Update README to include forking instructions before cloning repository

pbearne opened the pull request:
Trac ticket: https://core.trac.wordpress.org/ticket/64100

Commit:
Update README to include forking instructions before cloning repository

westonruter commented on README.md:
```diff
@@ -47,7 +47,11 @@ Ensure your container environment is running before using these commands.
 #### To start the development environment for the first time
 
-Clone the current repository using `git clone https://github.com/WordPress/wordpress-develop.git`. Then in your terminal move to the repository folder `cd wordpress-develop` and run the following commands:
+Goto https://github.com/WordPress/wordpress-develop and fork the repository to your own GitHub account. 
```
```suggestion
Go to https://github.com/WordPress/wordpress-develop and fork the repository to your own GitHub account. 
```

westonruter commented on README.md:
```diff
@@ -47,7 +47,11 @@ Ensure your container environment is running before using these commands.
+Goto https://github.com/WordPress/wordpress-develop and fork the repository to your own GitHub account. 
+
+Then clone the forked repository to your computer using `git clone https://github.com/<your-username>/wordpress-develop.git`.
+
```
```suggestion

Alternatively, if you have the GitHub CLI installed, you can just run `gh repo fork WordPress/wordpress-develop --clone --remote`. This will automatically add an `upstream` remote as well.

```

westonruter commented on README.md:
```diff
@@ -47,7 +47,11 @@ Ensure your container environment is running before using these commands.
-Clone the current repository using `git clone https://github.com/WordPress/wordpress-develop.git`. Then in your terminal move to the repository folder `cd wordpress-develop` and run the following commands:
+Goto https://github.com/WordPress/wordpress-develop and fork the repository to your own GitHub account. 
+
+Then clone the forked repository to your computer using `git clone https://github.com/<your-username>/wordpress-develop.git`.
```
```suggestion
Then clone the forked repository to your computer using `git clone https://github.com/<your-username>/wordpress-develop.git`. Add the origin repo as an `upstream` remote via `git remote add upstream https://github.com/WordPress/wordpress-develop.git`; then you can keep your branches up to date via `git pull --ff upstream/trunk`, for example.
```

pbearne committed:
Update README.md

Co-authored-by: Weston Ruter <westonruter@gmail.com>

pbearne committed:
Update README.md

Co-authored-by: Weston Ruter <westonruter@gmail.com>

pbearne committed:
Update README.md

Co-authored-by: Weston Ruter <westonruter@gmail.com>

westonruter commented on README.md:
```diff
@@ -47,7 +47,13 @@ Ensure your container environment is running before using these commands.
+
+Then clone the forked repository to your computer using `git clone https://github.com/<your-username>/wordpress-develop.git`. Add the origin repo as an `upstream` remote via `git remote add upstream https://github.com/WordPress/wordpress-develop.git`; then you can keep your branches up to date via `git pull --ff upstream/trunk`, for example.
+
+Alternatively, if you have the GitHub CLI installed, you can just run `gh repo fork WordPress/wordpress-develop --clone --remote`. This will automatically add an `upstream` remote as well.
```
```suggestion
Alternatively, if you have the [GitHub CLI](https://cli.github.com/) installed, you can just run `gh repo fork WordPress/wordpress-develop --clone --remote`. This will automatically add an `upstream` remote as well.
```

pbearne committed:
Update README.md

Co-authored-by: Weston Ruter <westonruter@gmail.com>

desrosj reviewed (changes requested):
I'm thinking that this would be better off as a numbered list of steps. Let's make the following changes:
- Add an intro sentence. "You can get started using the local development environment with these steps:"
- Where to go
- Click create fork
- Clone forked repo.
- Configure `upstream` origin
- How to keep your branches up to date.

After that, a include the "Alternatively" sentence.

And finally "After either if the two options above, move to the..."

pbearne committed:
Apply suggestion from @westonruter

Co-authored-by: Weston Ruter <westonruter@gmail.com>

pbearne committed:
Revise README for clearer setup instructions

Updated instructions for forking and cloning the repository with numbered steps for clarity.

desrosj commented on README.md:
```diff
@@ -47,7 +47,16 @@ Ensure your container environment is running before using these commands.
+
+1. Go to https://github.com/WordPress/wordpress-develop and fork the repository to your own GitHub account. 
+1. Then clone the forked repository to your computer using `git clone https://github.com/<your-username>/wordpress-develop.git`.
+1. Add the origin repo as an `upstream` remote via `git remote add upstream https://github.com/WordPress/wordpress-develop.git`;
```
```suggestion
1. Navigate into the directory for the cloned repository using `cd wordpress-develop`
1. Add the origin repo as an `upstream` remote via `git remote add upstream https://github.com/WordPress/wordpress-develop.git`;
```

Missing step. I believe you'll just get a "fatal: not a git repository" error without it.

desrosj commented on README.md:
```diff
@@ -47,7 +47,16 @@ Ensure your container environment is running before using these commands.
+1. Add the origin repo as an `upstream` remote via `git remote add upstream https://github.com/WordPress/wordpress-develop.git`;
+1. Then you can keep your branches up to date via `git pull --ff upstream/trunk`, for example.
+
+Alternatively, if you have the [GitHub CLI](https://cli.github.com/) installed, you can just run `gh repo fork WordPress/wordpress-develop --clone --remote`. This will automatically add an `upstream` remote as well.
```
```suggestion
Alternatively, if you have the [GitHub CLI](https://cli.github.com/) installed, you can simply run `gh repo fork WordPress/wordpress-develop --clone --remote` ([docs](https://cli.github.com/manual/gh_repo_fork)). This command will:
1. Fork the repository to your account (use the `--org` flag to clone into an organization).
1. Clone the repository to your machine. 
1, Add `WordPress/wordpress-develop` as `upstream` and set it to the default `remote` repository

After this, remember to run `cd wordpress-develop`.
```

desrosj commented on README.md:
```diff
@@ -47,7 +47,16 @@ Ensure your container environment is running before using these commands.
+1. Then you can keep your branches up to date via `git pull --ff upstream/trunk`, for example.
+
+Alternatively, if you have the [GitHub CLI](https://cli.github.com/) installed, you can just run `gh repo fork WordPress/wordpress-develop --clone --remote`. This will automatically add an `upstream` remote as well.
+
+After either of the two options above, move to the repository folder `cd wordpress-develop` and run the following commands in a terminal:
```
```suggestion
```

desrosj reviewed (changes requested):
Few more small suggestions. I think it's good to go after those, though. I will commit tomorrow unless there are any objections.

pbearne committed:
Apply suggestions from code review

Co-authored-by: Jonathan Desrosiers <359867+desrosj@users.noreply.github.com>

peterwilsoncc commented:
This looks good to me for the purpose of clarifying to create a fork.

While editing the file, should the code spaces reference be removed as they haven't worked for a while. See https://core.trac.wordpress.org/ticket/57896

<img width="549" height="230" alt="Screenshot 2025-11-11 at 10 26 39 am" src="https://github.com/user-attachments/assets/e5341263-319d-4a8d-a164-534955c96459" />

t-hamano commented on README.md:
```diff
@@ -47,7 +47,22 @@ Ensure your container environment is running before using these commands.
+Alternatively, if you have the [GitHub CLI](https://cli.github.com/) installed, you can simply run `gh repo fork WordPress/wordpress-develop --clone --remote` ([docs](https://cli.github.com/manual/gh_repo_fork)). This command will:
+1. Fork the repository to your account (use the `--org` flag to clone into an organization).
+1. Clone the repository to your machine. 
+1, Add `WordPress/wordpress-develop` as `upstream` and set it to the default `remote` repository
```
```suggestion
1. Go to https://github.com/WordPress/wordpress-develop and fork the repository to your own GitHub account. 
1. Then clone the forked repository to your computer using `git clone https://github.com/<your-username>/wordpress-develop.git`.
1. Navigate into the directory for the cloned repository using `cd wordpress-develop`.
1. Add the origin repo as an `upstream` remote via `git remote add upstream https://github.com/WordPress/wordpress-develop.git`.
1. Then you can keep your branches up to date via `git pull --ff upstream/trunk`, for example.

Alternatively, if you have the [GitHub CLI](https://cli.github.com/) installed, you can simply run `gh repo fork WordPress/wordpress-develop --clone --remote` ([docs](https://cli.github.com/manual/gh_repo_fork)). This command will:
1. Fork the repository to your account (use the `--org` flag to clone into an organization).
1. Clone the repository to your machine. 
1. Add `WordPress/wordpress-develop` as `upstream` and set it to the default `remote` repository
```

desrosj committed:
Correct punctuation.

Co-authored-by: Aki Hamano <54422211+t-hamano@users.noreply.github.com>

desrosj commented:
> While editing the file, should the code spaces reference be removed as they haven't worked for a while. See https://core.trac.wordpress.org/ticket/57896

That makes sense, I think. It would be great to get that fixed, but preventing frustration for anyone attempting to use that as a means to contribute is probably best in the mean time. I'll remove that section when committing.
